orderer/common/multichannel/registrar.go

```diff
@@ -370,3 +371,13 @@ func (r *Registrar) NewChannelConfig(envConfigUpdate *cb.Envelope) (channelconfi
 func (r *Registrar) CreateBundle(channelID string, config *cb.Config) (channelconfig.Resources, error) {
 	return channelconfig.NewBundle(channelID, config, r.bccsp)
 }
+
+func (r *Registrar) ListAllChannels() ([]channelparticipation.ChannelInfoShort, string) {
+	//TODO
+	return make([]channelparticipation.ChannelInfoShort,0), ""
+}
+
+func (r *Registrar) ListChannel(channelID string) (*channelparticipation.ChannelInfoFull, error) {
```

Comment:
Perhaps indicate the system channel string into the struct returned from this API as a boolean?

Reply:
We want to quickly identify whether we have one or not w/o scanning the entire list... this goes to the client so we want this efficient and clear. Note that the system channel may change from "" to "i-am-here" and back to "" as we can create and delete it with this API.

Reply:
Under what circumstance do we need to identify if we have a system channel or not, and not also anyway return all channels? 

The way this is used in the existing code in this PR is that you return all the channels anyway, so you are going to see who is the system channel when you iterate over it anyway, no?

Reply:
Yes, but this structure is intended mainly for the client, who receives JSON.
The client may want to identify the system channel immediately, for example, when it needs to know if it is at all possible to create channels or only list them. If the system channel exists, the API can only serve GET and DELETE of the system channel. This way, the client can find that out without iterating.

Reply:
Moreover, this representation is shorter. 

`{
  "size":3,
  "systemChannel":"a",
  "channels":[
     {"name":"a","url":"/api/channels/a"},
     {"name":"b","url":"/api/channels/b"},
     {"name":"c","url":"/api/channels/c"},
]}`

is shorter than 

`{
  "size":2,
  "channels":[
     {"name":"a","url":"/api/channels/a","systemChannel":"true"}, 
     {"name":"b","url":"/api/channels/b","systemChannel":"false"},
     {"name":"c","url":"/api/channels/c","systemChannel":"false"},
]}`

even for a short list.

Reply:
The other alternative I see is:

```
{
  "systemChannel":{"name":"a","url":"/api/channels/a"},
  "size":2,
  "channels":[
     {"name":"b","url":"/api/channels/b"},
     {"name":"c","url":"/api/channels/c"},
]}
```
i.e. that the `channels` array carry only application channels.

Reply:
I like the last one the most, modulo the size which is redundant.

Reply:
There is another option to return a map:

```
{
  "systemChannel": "a",
  "channels": {
      "a": "/api/channels/a",
      "b": "/api/channels/b",
      "c": "/api/channels/c",
   }
}
```

Reply:
OK Lets's go with 
```
{
  "systemChannel":{"name":"a","url":"/api/channels/a"},
  "channels":[
     {"name":"b","url":"/api/channels/b"},
     {"name":"c","url":"/api/channels/c"},
]}
```
I want to keep the ChannelInfoShort struct just in case we'll need to add another field one day.